website/docs/developer/research/research-deep-glacier-migration.md

```diff
@@ -0,0 +1,154 @@
+One option of bulk retrieval is to use python boto3 client for S3:
+
+```python
+import boto3
+
+#this objects list should be provided by the user
+retrieve_objects = ['test1.xml', 'test2.xml']
+
+s3 = boto3.client('s3')
+for retrieve_object in retrieve_objects:
+    try:
+        response = s3.restore_object(
+                Bucket='orca-rhassan-sandbox-glacier', # bucket that contains the objects currently in glacier flexible retrieval
+                Key = retrieve_object,
+                RestoreRequest={
+                    'Days': 30,
+                    'GlacierJobParameters': {
+                        'Tier': 'Bulk',
+                    },
+                },
+            )
+        print(response)
+    except Exception as ex:
+        print(f"{ex} for {retrieve_object}")
+
+```
+Output:
+```
+{'ResponseMetadata': {'RequestId': '61ZWC27NHSFNMYJT', 'HostId': 'ItKZaYvn9rfi6ZVYmUjfZX24mKJlDZlciT9MaKK0aHiNuLWB3Pt8WSEMNt5yBK+5u+MJfzjFDfU=', 'HTTPStatusCode': 202, 'HTTPHeaders': {'x-amz-id-2': 'ItKZaYvn9rfi6ZVYmUjfZX24mKJlDZlciT9MaKK0aHiNuLWB3Pt8WSEMNt5yBK+5u+MJfzjFDfU=', 'x-amz-request-id': '61ZWC27NHSFNMYJT', 'date': 'Sun, 24 Jul 2022 18:01:33 GMT', 'server': 'AmazonS3', 'content-length': '0'}, 'RetryAttempts': 0}}
+{'ResponseMetadata': {'RequestId': 'PNN4XXA5SYTBBRKG', 'HostId': 'HgiIiqfb0hZPHgZml98VCyLLfD5LyQ8pSLpgGN6hwUUMtmcPZoo/ACbCL1rXz+pXZ4Ce2UEe34s=', 'HTTPStatusCode': 202, 'HTTPHeaders': {'x-amz-id-2': 'HgiIiqfb0hZPHgZml98VCyLLfD5LyQ8pSLpgGN6hwUUMtmcPZoo/ACbCL1rXz+pXZ4Ce2UEe34s=', 'x-amz-request-id': 'PNN4XXA5SYTBBRKG', 'date': 'Sun, 24 Jul 2022 18:01:34 GMT', 'server': 'AmazonS3', 'content-length': '0'}, 'RetryAttempts': 0}}
+```
+
+Also see existing ORCA [restore_object function](https://github.com/nasa/cumulus-orca/blob/develop/tasks/request_files/request_files.py#L548)
+Make sure the S3 bucket policy has permission for `s3:PutObject*` action otherwise user will see access denied error. See this [policy](https://github.com/nasa/cumulus-orca/blob/develop/website/docs/developer/deployment-guide/creating-orca-glacier-bucket.md#archive-bucket) for additional details.
+
+Once all files are restored from glacier, there should be some mechanism that will notify users that the retrieval process is complete. One approach could be to enable event notifications in bucket that will trigger either an SQS, SNS or lambda. This will require additional research since AWS notifies for each `s3:ObjectRestore:Completed` API call and not for a specific number of objects.
+
+- Once all objects are retrieved in standard S3 storage, the bucket's lifecycle policy can be configured with `Deep Archive` storage type if the whole bucket objects have to be moved to deep archive.
```

Comment:
Why do we need to restore them first? According to docs ["You can transition ... Any storage class to the S3 Glacier Deep Archive storage class."](https://docs.aws.amazon.com/AmazonS3/latest/userguide/lifecycle-transition-general-considerations.html)

Reply:
yes that was my initial idea to use lifecycle rule in the bucket.

However, the end user might only want to restore some specific files in the bucket. In that case, the lifecycle rule will not work because we don't know what type of files the user might request.

Reply:
Per @AndrewDorn comment, you really have two use cases here you should break out and discuss in better clarity. Each may be a slightly diffrent design approach.
- The first is where a user wants to either convert all the holdings or holdings with a know key prefix to DEEP GLACIER.
- The second is where a user wants to convert specific collection or granules to DEEP Glacier which may not have the same key prefix or are intermixed with granules that share the key prefix.

Reply:
We are in the section about moving all files in the bucket.

Reply:
broken down into two separate approach.